Mcreator 2025.3 gradle fails to build on 2 NeoForge workspaces on Linux.

### Issue tracker rule checks (read carefully)

- [x] I have checked that my problem <a href='https://github.com/MCreator/MCreator/issues?q=is%3Aissue' target='_blank'>is not already reported</a>
- [x] I have checked that my problem is not covered on <a href='https://mcreator.net/support/knowledgebase' target='_blank'>Knowledge Base</a> or on <a href='https://mcreator.net/wiki' target='_blank'>MCreator's Wiki</a>
- [x] I have made sure this bug is not plugin-related (tested with MCreator without plugins installed)
- [x] I have checked that my written content does not violate the <a href='https://mcreator.net/wiki/general-publishing-guidelines' target='_blank'>publishing guidelines</a>

### Issue description

I seem to have an issue that on trying to import two 2025.2 workspaces, a texture pack mod and a mod, both for NeoForge 1.24.1, the error that I receive the following errors across both workspaces has a similar log that is listed under the logs section of the report.

I have already rebooted my computer several times after clearing the gradle file across both mods for it to build initially as well.

### How to reproduce this issue?

Step 1: Open MCreator 2025.3 after installation.
Step 2: import a 1.24.1 NeoForge workspace from 2025.2
Error given of workspace setup failed.

### Operating system

Linux

### Details

MCreator Version: 2025.3 (Release)
Generator: NeoForge 1.28.1

### Example workspace

_No response_

### Logs

Executing Gradle synchronization tasks
Build info: MCreator 2025.3.44519, neoforge-1.21.8, 64-bit, 47456 MB, Linux, JVM 21.0.8, JAVA_HOME: /home/wfdplus/Downloads/MCreator20253/jdk, started on: 2025-10-31-19:43:09

org.gradle.tooling.GradleConnectionException: Could not run phased build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-8.14.2-bin.zip'.
 at org.gradle.tooling.internal.consumer.ConnectionExceptionTransformer.transform(ConnectionExceptionTransformer.java:73 at org.gradle.tooling.internal.consumer.ConnectionExceptionTransformer.transform(ConnectionExceptionTransformer.java:73)
 at org.gradle.tooling.internal.consumer.ResultHandlerAdapter.onFailure(ResultHandlerAdapter.java:42 at org.gradle.tooling.internal.consumer.ResultHandlerAdapter.onFailure(ResultHandlerAdapter.java:42)
 at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor$1$1.run(DefaultAsyncConsumerActionExecutor.java:68 at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor$1$1.run(DefaultAsyncConsumerActionExecutor.java:68)
 at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64 at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
 at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47 at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
 at java.base/java.lang.Thread.run(Thread.java:1583 at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.gradle.process.internal.ExecException: A problem occurred starting process 'Gradle build daemon'
 at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:241 at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:241)
 at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:218 at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:218)
 at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:396 at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:396)
 at org.gradle.process.internal.ExecHandleRunner.lambda$run$3(ExecHandleRunner.java:102 at org.gradle.process.internal.ExecHandleRunner.lambda$run$3(ExecHandleRunner.java:102)
 at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85 at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85)
 at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:101 at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:101)
 at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64 at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
 at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48 at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
 ... 3 more
Caused by: net.rubygrapefruit.platform.NativeException: Could not start '/home/wfdplus/Downloads/MCreator20253/jdk/bin/java'
 at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27 at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
 at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36 at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36)
 at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:122 at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:122)
 at org.gradle.process.internal.ExecHandleRunner.lambda$run$0(ExecHandleRunner.java:80 at org.gradle.process.internal.ExecHandleRunner.lambda$run$0(ExecHandleRunner.java:80)
 at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85 at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85)
 at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:79 at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:79)
 ... 5 more
Caused by: java.io.IOException: Cannot run program "/home/wfdplus/Downloads/MCreator20253/jdk/bin/java" (in directory "/home/wfdplus/.mcreator/gradle/daemon/8.14.2"): error=13, Permission denied
 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1170 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1170)
 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1089 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1089)
 at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25 at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
 ... 10 more
Caused by: java.io.IOException: error=13, Permission denied
 at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
 at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:295 at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:295)
 at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:225 at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:225)
 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1126 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1126)
 ... 12 more

TASK EXECUTION FAILED
Task completed in less than a second



Executing Gradle synchronization tasks
Build info: MCreator 2025.3.44519, resourcepack-1.21.8, 64-bit, 47456 MB, Linux, JVM 21.0.8, JAVA_HOME: /home/wfdplus/Downloads/MCreator20253/jdk, started on: 2025-10-31-20:09:10

org.gradle.tooling.GradleConnectionException: Could not run phased build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-9.1.0-bin.zip'.
 at org.gradle.tooling.internal.consumer.ConnectionExceptionTransformer.transform(ConnectionExceptionTransformer.java:73 at org.gradle.tooling.internal.consumer.ConnectionExceptionTransformer.transform(ConnectionExceptionTransformer.java:73)
 at org.gradle.tooling.internal.consumer.ResultHandlerAdapter.onFailure(ResultHandlerAdapter.java:42 at org.gradle.tooling.internal.consumer.ResultHandlerAdapter.onFailure(ResultHandlerAdapter.java:42)
 at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor$1$1.run(DefaultAsyncConsumerActionExecutor.java:68 at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor$1$1.run(DefaultAsyncConsumerActionExecutor.java:68)
 at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64 at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
 at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47 at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
 at java.base/java.lang.Thread.run(Thread.java:1583 at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.gradle.process.ProcessExecutionException: A problem occurred starting process 'Gradle build daemon'
 at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:242 at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:242)
 at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:219 at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:219)
 at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:404 at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:404)
 at org.gradle.process.internal.ExecHandleRunner.lambda$run$3(ExecHandleRunner.java:132 at org.gradle.process.internal.ExecHandleRunner.lambda$run$3(ExecHandleRunner.java:132)
 at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:84 at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:84)
 at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:131 at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:131)
 at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64 at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
 at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47 at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
 ... 3 more
Caused by: net.rubygrapefruit.platform.NativeException: Could not start '/home/wfdplus/Downloads/MCreator20253/jdk/bin/java'
 at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27 at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
 at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:152 at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:152)
 at org.gradle.process.internal.ExecHandleRunner.lambda$run$0(ExecHandleRunner.java:110 at org.gradle.process.internal.ExecHandleRunner.lambda$run$0(ExecHandleRunner.java:110)
 at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:84 at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:84)
 at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:109 at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:109)
 ... 5 more
Caused by: java.io.IOException: Cannot run program "/home/wfdplus/Downloads/MCreator20253/jdk/bin/java" (in directory "/home/wfdplus/.mcreator/gradle/daemon/9.1.0"): error=13, Permission denied
 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1170 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1170)
 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1089 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1089)
 at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25 at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
 ... 9 more
Caused by: java.io.IOException: error=13, Permission denied
 at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
 at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:295 at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:295)
 at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:225 at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:225)
 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1126 at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1126)
 ... 11 more

TASK EXECUTION FAILED
Task completed in 18 seconds

--- a/platform/linux/linux.gradle
+++ b/platform/linux/linux.gradle
@@ -28,34 +28,20 @@ tasks.register('exportLinux64', Copy.class) {
 tasks.register('tarLinux64', Tar.class) {
     dependsOn exportLinux64
 
-    into('MCreator' + (String) project.mcreatorconf.getProperty('mcreator').replace(".", "").replace(" ", "")) {
-        from('build/export/linux64/') { exclude "mcreator.sh", "jdk/bin*" }
-        from('build/export/linux64/mcreator.sh') {
-            filePermissions {
-                user {
-                    read = true; write = true; execute = true
-                }
-                group {
-                    read = true; execute = true
-                }
-                other {
-                    read = true; execute = true
-                }
-            }
+    def execPermissions = {
+        filePermissions {
+            user { read = true; write = true; execute = true }
+            group { read = true; execute = true }
+            other { read = true; execute = true }
         }
-        from('build/export/linux64/jdk/bin/') {
-            filePermissions {
-                user {
-                    read = true; write = true; execute = true
-                }
-                group {
-                    read = true; execute = true
-                }
-                other {
-                    read = true; execute = true
-                }
-            }
-        } into 'jdk/bin/'
+    }
+
+    into('MCreator' + (String) project.mcreatorconf.getProperty('mcreator').replace(".", "").replace(" ", "")) {
+        from('build/export/linux64/') { exclude "mcreator.sh", "jdk/bin*", "jdk/lib/jspawnhelper", "jdk/lib/jexec" }
+        from('build/export/linux64/mcreator.sh', execPermissions)
+        from('build/export/linux64/jdk/bin/', execPermissions) into 'jdk/bin/'
+        from('build/export/linux64/jdk/lib/jspawnhelper', execPermissions) into 'jdk/lib/'
+        from('build/export/linux64/jdk/lib/jexec', execPermissions) into 'jdk/lib/'
     }
 
     destinationDirectory = file('build/export/')

